lib/utils.js

```diff
@@ -0,0 +1,22 @@
+const api = require('./heroku-api')
+const cli = require('heroku-cli-util')
+
+function * getPipeline (context, client) {
+  let pipeline = context.flags.pipeline
+
+  let pipelineOrApp = pipeline || context.app
+  if (!pipelineOrApp) cli.exit(1, 'Required flag:  --pipeline PIPELINE or --app APP')
+
+  if (pipeline) {
+    pipeline = yield api.pipelineInfo(client, pipeline)
```

Comment:
I think this will need to handle the case where the user has >1 pipeline with the same name. In that instance the API returns a [HTTP 300](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/300).

In the [pipelines CLI](https://github.com/heroku/heroku-pipelines/blob/master/lib/disambiguate.js) we make a different request to `GET /pipelines?eq[name]=name` and provide a disambiguation UI if there are > 1 with that name.

Hopefully you can add `heroku-pipelines` as a dependency of this plugin and reuse the same code so you don't have to reimplement it here.